test/tools/ossfuzz/SolidityGenerator.h

```diff
@@ -0,0 +1,44 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Implements generators for synthesizing mostly syntactically valid
+ * Solidity test programs.
+ */
+
+#pragma once
+
+#include <libsolutil/Whiskers.h>
+
+#include <random>
+
+namespace solidity::test::fuzzer
+{
+using RandomEngine = std::mt19937_64;
+
+class SolidityGenerator
+{
+public:
+	SolidityGenerator(unsigned int _seed): m_rand(_seed)
+	{}
+	/// @returns a pseudo randomly generated test program
+	std::string visit();
```

Comment:
Any particular reason why the function is called `visit`? How about `operator()`?

Reply:
Hmm... Yeah, visit is undescriptive. Perhaps `generateTestCase`?

Reply:
Sounds good